src/Model/Profile.php

```diff
@@ -1133,6 +1119,43 @@ public static function openWebAuthInit($token)
 
 			$_SESSION['remote'][] = ['cid' => $contact['id'], 'uid' => $contact['uid'], 'url' => $visitor['url']];
 		}
+
+		$a->contact = $visitor;
```

Comment:
As we currently call this method out of app, I'd prefer returning the visitor and set it as `$this->contact` inside `App`.

Reply:
This is just a copy from the already existing code from the function "openWebAuthInit" in this file. I wanted to only apply the needed changes and not to restructure everything.

Reply:
I know this kind of problem ;-)
OK, at least I'd like to add a `@todo` for later. This is something I'd like to generally avoid in the code
`App` -> `AnotherClass` -> `App` dependency.

Reply:
BTW: This is some other "I don't really understand the problem" thing ;-) Have I said that you have to explain me more of this in some meeting, possibly at the Hackathon in Berlin? :grinning:

Reply:
I'll do my best to join at the Hackathon :-)